lib/Interpreter/CIFactory.cpp

```diff
@@ -893,16 +902,64 @@ namespace {
     // of the two is the underlying compiler.
 
 #ifdef __clang__
-    PPOpts.addMacroDef("__CLING__clang__=" ClingStringify(__clang__));
+  #if defined(__apple_build_version__)
+     // __apple_build_version__ >= 8000031 Xcode 8
+     #if __apple_build_version__ >= 7030029
```

Comment:
More arguments against this:
- nobody should check against the clang version, but against compiler features
- I can almost guarantee that we will forget to maintain this in the future.

I see that Xcode 8 defines the clang version as 8.0.0.38, not mentioning any version number of the underlying clang. That's the versioning mess that I was referring to - and I do not see it as cling's task to fix that :-)